test/testHelpers.hpp

```diff
@@ -242,6 +242,11 @@ bool noHalfTests(af::dtype ty);
     if (noHalfTests((af_dtype)af::dtype_traits<type>::af_type))   \
     GTEST_SKIP() << "Device doesn't support Half"
 
+static af_err CHECK_SUPPORTED_ERR_RETURN;
+#define CHECK_SUPPORTED(af_stat) \
+    if (af_stat == AF_ERR_NOT_SUPPORTED) \
+        GTEST_SKIP() << "Function not supported on this backend"
```

Comment:
would be good to print the backend name.

Reply:
I've looked into this and an error message with the back end name is generated in backend/common/err_common.cpp in the set_global_error_string function. However it is only printed if the AF_PRINT_ERRORS environment variable is set. Not sure the reason for this.

Reply:
I also notice that in the ***_NOT_SUPPORTED macros a string is being passed called 'message' however this string is then assigned to the 'backend' member of the SupportError class. So when getBackendName() is called the 'message' is being returned instead of the back end name.

Reply:
It is difficult to obtain the back end name at this point in the code. If a C++ exception is triggered it will print out the back end as part of the verbose error message when the AF_PRINT_ERRORS environment variable is set. For the C API you just get a AF_ERR_NOT_SUPPORTED returned from the function.